### PR TITLE
Cache `~/.cocoapods` on CircleCI

### LINKED ISSUE #450: Speed up `pod lib lint` job on CI
Currently takes up the most time by far of all the things we do on CI because we clone all of CocoaPods' Specs repo, which seems extremely inefficient considering that we only rely on two dependencies (SWXMLHash & Yams).

### PR DESCRIPTION
**Pull request by norio-nomura**
Trying to resolve #450 by using [Caching Dependencies](https://circleci.com/docs/2.0/caching/).

### CODE REVIEW

**Comment by norio-nomura** [comment]
First invocation: https://circleci.com/workflow-run/25360d9f-5087-4451-9a77-ce006a0a0ec6
screenshot 2017-11-30 11 39 44

Second invocation: https://circleci.com/workflow-run/19bd48b1-f053-4872-9f02-e317ad69d93c
screenshot 2017-11-30 11 40 53

It seems to be worth using cache.

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,7 +10,14 @@ jobs:
       - run: make test|xcpretty -r junit
       - store_test_results:
           path: build/reports/
+      - restore_cache:
+          keys:
+            - cocoapods-{{ .Environment.CIRCLE_JOB }}
       - run: pod lib lint
+      - save_cache:
+          key: cocoapods-{{ .Environment.CIRCLE_JOB }}
```
Comment (jpsim) [inline .circleci/config.yml]: Why does this need to be keyed by the circle job? Why not just `key: cocoapods`?

Comment (norio-nomura) [reply .circleci/config.yml]: I was worried that conflicts might occur if the same key is used for two jobs.
I don't know whether that is required or not.

Comment (norio-nomura) [reply .circleci/config.yml]: I changed the key to `cocoapods`, but it seems to be okay. 👍

**Comment by jpsim** [comment]
Excellent work!

**Pull request merged**